### PR TITLE
std.crypto.tls: implement TLSv1.2

### PR DESCRIPTION
**Pull request by jacobly0**
Tested with ziglang/contrib-testing@ec35cfdbb8296b1cde82a7ef5e7e4462c649feaa.

### CODE REVIEW

**Comment by andrewrk** [comment]
related:
* #14172
* #17213

**Comment by jacobly0** [comment]
According to [top_sites.zig](https://github.com/jacobly0/tls-conformance/blob/595b78481937a9f5b0380b3421e245d369839a74/top_sites.zig), this PR goes from `228/500 succeeded` to `260/500 succeeded`.

File: lib/std/crypto/25519/ed25519.zig [hunk 1]
```diff
@@ -214,17 +219,26 @@ pub const Ed25519 = struct {
...
-            st.update(msg);
-            return st.verify();
+        pub fn verify(sig: Signature, msg: []const u8, public_key: PublicKey) VerifyError!void {
+            try sig.concatVerify(&.{msg}, public_key);
+        }
+
+        /// Verify the signature against a concatenated message and public key.
+        /// Return IdentityElement or NonCanonical if the public key or signature are not in the expected range,
+        /// or SignatureVerificationError if the signature is invalid for the given message and key.
+        pub fn concatVerify(sig: Signature, msg: []const []const u8, public_key: PublicKey) VerifyError!void {
```
Comment (jedisct1) [inline lib/std/crypto/25519/ed25519.zig:238]: As this is very protocol specific, maybe the `tis` namespace would be a better place for that function?

**Review by andrewrk — APPROVED** [review]

**Comment by jacobly0** [comment]
Now at `358/500 succeeded` with `x25519_ml_kem768` key share fixed, not including up to 2 flaky regressions.

**Comment by jacobly0** [comment]
Now at `374/500 succeeded`.

File: lib/std/std.zig [hunk 2]
```diff
@@ -146,6 +146,11 @@ pub const Options = struct {
     /// make a HTTPS connection.
     http_disable_tls: bool = false,
 
+    /// This enables `std.http.Client` to log ssl secrets to the file specified by the SSLKEYLOGFILE
+    /// env var.  Creating such a log file allows other programs with access to that file to decrypt
+    /// all `std.http.Client` traffic made by this program.
+    http_enable_ssl_key_log_file: bool = @import("builtin").mode == .Debug,
```
Comment (karlseguin) [inline lib/std/std.zig:152]: Maybe this should default to false in all builds?

Comment (andrewrk) [reply lib/std/std.zig:152]: Note that it also requires the env var to be set. So for the secrets to be logged the user (or parent process) has to explicitly ask for it. In both cases the secrets could have been accessed via other mechanisms (e.g. ptrace).

**Comment by jacobly0** [comment]
I don't see any way to increase compatibility with these buggy tls implementations without removing features, so that's where I leave it for now.

**Pull request merged**